app/models/course/assessment/question.rb

```diff
@@ -5,4 +5,17 @@ class Course::Assessment::Question < ActiveRecord::Base
   has_and_belongs_to_many :tags
 
   delegate :to_partial_path, to: :actable
+
+  # Attempts the given question in the submission. This builds a new answer for the current
+  # question.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which the answer should
+  #   belong to.
+  # @return [Course::Assessment::Answer] The answer corresponding to the question. It is required
+  #   that the {Course::Assessment::Answer#question} property be the same as +self+. The result
+  #   should not be persisted.
+  def attempt(submission)
```

Comment:
I have the complain just because that when I read the code, it looks like question attempts a submission, not question be attempted by student. Then I have to read your documentation and implementation here to figure out why